File: Java/p10341.java
import java.util.Scanner;
/**
 * Write a description of class p10341 here.
 * 
 * @author (your name) 
 * @version (a version number or a date)
 */
public class p10341
{
    public static void main (String [] args) {
        Scanner input = new Scanner(System.in);
        while (input.hasNext()) {
            double p = input.nextDouble();
            double q = input.nextDouble();
            double r = input.nextDouble();
            double s = input.nextDouble();
            double t = input.nextDouble();
            double u = input.nextDouble();
            
            double res = bisectionMethod(0, 1, p, q, r, s, t, u);
            if ( res == -1) System.out.println("No solution");
            else System.out.printf("%.4f\n", res);
        }
    }
    
    static double bisectionMethod (double low, double high, double p, double q, double r, double s, double t, double u) {
        if (function(low, p, q, r, s, t, u) * function(high, p, q, r, s, t, u) > 0) return -1;
        else {
            double mid = (low + high) / 2;
            double function = function(mid, p, q, r, s, t, u);
            if (function > -0.0000001 && function < 0.0000001) {
                return mid;
            } else if (signCheck(function(low, p, q, r, s, t, u), function)) {
                return bisectionMethod(mid, high, p, q, r, s, t, u);
            } else {
                return bisectionMethod(low, mid, p, q, r, s, t, u);
            }
        }
    }
    
    static double function (double x, double p, double q, double r, double s, double t, double u) {
        return
            (p * (Math.pow(Math.E, -x))) +
            (q * (Math.sin(x))) +
            (r * (Math.cos(x))) +
            (s * (Math.tan(x))) +
            (t * (Math.pow(x, 2))) +
            u;
    }
    
    static boolean signCheck (double a, double b) {
        return (a >= 0 && b >= 0) || (a < 0 && b < 0);
    }
}

File: C,C++/p10341.c
#include<stdio.h>
#include<math.h>
#define E (2.7182818284590452353602874713526624977572470937L )

double bisectionMethod (double low, double high, double p, double q, double r, double s, double t, double u);
double function (double x, double p, double q, double r, double s, double t, double u);
int signCheck (double a, double b);

int main()
{
    double p,q,r,s,t,u;
    while (scanf("%lf %lf %lf %lf %lf %lf", &p, &q, &r, &s, &t, &u) != EOF) {
    	double res = bisectionMethod(0.0,1.0,p,q,r,s,t,u);
    	if (res == -1) printf("No solution\n");
    	else printf("%.4f\n", res);
	}

    return 0;
}

double bisectionMethod (double low, double high, double p, double q, double r, double s, double t, double u) {
	if (function(low, p, q, r, s, t, u) * function(high, p, q, r, s, t, u) > 0) return -1;
        else {
            double mid = (low + high) / 2.0;
            double function1 = function(mid, p, q, r, s, t, u);
            if (function1 > -0.0000001 && function1 < 0.0000001) {
                return mid;
            } else if (signCheck(function(low, p, q, r, s, t, u), function1)) {
                return bisectionMethod(mid, high, p, q, r, s, t, u);
            } else {
                return bisectionMethod(low, mid, p, q, r, s, t, u);
            }
        }
}
    
double function (double x, double p, double q, double r, double s, double t, double u) {
	return
            (p * (pow(E, -x))) +
            (q * (sin(x))) +
            (r * (cos(x))) +
            (s * (tan(x))) +
            (t * (pow(x, 2))) +
            u;
}

int signCheck (double a, double b) {
	if ( (a >= 0 && b >= 0) || (a < 0 && b < 0)) return 1;
	else return 0;
}
